core/src/hash/hasher.rs

```diff
@@ -0,0 +1,62 @@
+// Copyright 2018-2020 Parity Technologies (UK) Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! Supported cryptographic hashing algorithms.
+
+/// Types that implement this trait are marker types that identify a supported
+/// cryptographic hash function.
+pub trait Hasher {
+    /// The output of the hash function.
+    ///
+    /// # Note
+    ///
+    /// This is a byte slice with varying lengths, e.g. `[u8; 32]`, [`u8; 16]`, etc.
+    type Output: Default;
+
+    /// Finalizes the hash using the underlying procedure.
+    fn finalize_immediate(input: &[u8], output: &mut Self::Output);
+}
+
+macro_rules! impl_hasher_for {
+    (
+        $( #[$doc:meta] )*
```

Comment:
This throws away the documentation comments, which is bad.

Reply:
good catch!

Reply:
Thank you!